examples/darksage.py

```diff
@@ -3569,88 +3569,93 @@ def jH2(self, tree, j_bin):
     def RadiusHI(self, tree, h):
         DiscRadii = np.zeros((len(tree),30))
         SigmaHI = np.zeros((len(tree),30))
+        arr = np.zeros(len(tree))
         for i in range(1,31):
             DiscRadii[:,i-1] = (tree['DiscRadii_'+str(i)]+tree['DiscRadii_'+str(i-1)])*0.5
             SigmaHI[:,i-1] = tree['DiscHI_'+str(i)]/(np.pi*(tree['DiscRadii_'+str(i)]**2-tree['DiscRadii_'+str(i-1)]**2))*1e-2*h
         (row, col) = np.where(SigmaHI>1.0)
         filt = np.append(np.diff(row)>0, True)
-        row, col = row[filt], col[filt]
-        arr = np.zeros(len(tree))
-        arr[row] = DiscRadii[row,col]
-        row, col = row[col<29], col[col<29]
-        arr[row] = arr[row] + (SigmaHI[row,col]-1.0)/(SigmaHI[row,col]-SigmaHI[row,col+1]) * (DiscRadii[row,col+1]-DiscRadii[row,col])
+        if len(filt[filt])>0:
+            row, col = row[filt], col[filt]
```

Comment:
This line is still crashing while running on tree file 104 (starting at index 0). 

```
Working on files written by cpu #102
Done with 102. Time taken = 1386.21046782 seconds. Ntrees converted = 2849599 (out of 14329882). Cumulative time = 123496.881334 seconds.
Working on files written by cpu #103
Done with 103. Time taken = 1184.85756302 seconds. Ntrees converted = 2874291 (out of 14329882). Cumulative time = 124681.738897 seconds.
Working on files written by cpu #104
Traceback (most recent call last):
  File "/home/msinha/.local/bin/taoconvert", line 4, in <module>
    __import__('pkg_resources').run_script('taoconvert==0.1', 'taoconvert')
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 750, in run_script
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 1527, in run_script
  File "/home/msinha/.local/lib/python2.7/site-packages/taoconvert-0.1-py2.7.egg/EGG-INFO/scripts/taoconvert", line 87, in <module>
    converter.convert()
  File "/home/msinha/.local/lib/python2.7/site-packages/taoconvert-0.1-py2.7.egg/tao/Converter.py", line 207, in convert
    for tree in self.iterate_trees():
  File "/home/msinha/research/codes/TAOImport/examples/darksage.py", line 3906, in iterate_trees
    tree[fieldname] = conversion_function(tree, h)
  File "/home/msinha/research/codes/TAOImport/examples/darksage.py", line 3553, in RadiusHI
    row, col = row[filt], col[filt]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 0 but corresponding boolean dimension is 1

```

Reply:
Right, this was clearly flawed.  Have now updated.

Reply:
Able to get things going with that edit?

Reply:
The converter is running since yesterday. Unsure if the tree # 103 has been processed yet. Last logged output was 5 hours ago and tree # 86 was getting processed (ozstar seems to be buffering the logging)

Reply:
Past the tree that was failing

Reply:
Woot!

Reply:
How's the status?  Get to the end without crashing?

Reply:
Sadly no. Timed out after 7 days, had finished only 400 something files out of the 512. I have to now investigate the MPI option for the converter

Reply:
It's like the Universe doesn't want us to succeed.  What's the bottleneck?  The computed fields?  Could that be solved by modifying Dark Sage to calculate them and store them?

Reply:
I'll be on leave from Wednesday -- is there anything I can do to help get Dark Sage live again?  Are there interactive nodes on Ozstar that can be used to circumvent the time limit for jobs?

Reply:
It's fine for the moment - I am trying to figure out a couple of ways to run the Darksage converter to completion. 

Separately, I am also not planning to merge this PR in until we have the Darksage 2016 model up and running. Or we can just sort the merge out in person when I am over in Perth the end of the month